### PR TITLE
Login: remove signup resume code

### PR DESCRIPTION
**Pull request by jsnajdr**
Removes code that resumes the last signup step after successful login. Originally added in #34770, the mostly removed in #35552. This PR removes the last bit of the no-longer-needed code, also removing a signup state dependency from the login entrypoint.

Makes #33642 (login entrypoint) mergeable.

### CODE REVIEW

File: client/blocks/login/index.jsx [hunk 1]
```diff
@@ -145,33 +143,23 @@ class Login extends Component {
...
 			two_factor_enabled: this.props.twoFactorEnabled,
 			social_service_connected: this.props.socialConnect,
 		} );
 
 		// Redirects to / if no redirect url is available
-		const url = redirectTo ? redirectTo : window.location.origin;
+		const url = this.props.redirectTo || '/';
 
 		// User data is persisted in localstorage at `lib/user/user` line 157.
 		// Only clear the data if a user is currently set, otherwise keep the
```
Comment (sgomes) [inline client/blocks/login/index.jsx]: Is the statement in this comment still true? In particular, does signup reuse it?

**Review by sgomes — APPROVED** [review]
Thank you, @jsnajdr! The code looks good, and this appears to work correctly.

Left a minor comment, and it looks like this is triggering some lint errors because of the recent deprecation of `createReducer`, but I'm happy for you to ignore that.

Any idea why this code wasn't removed when the A/B test was?

File: client/blocks/login/index.jsx (hunk 1 above)
Comment (jsnajdr) [reply client/blocks/login/index.jsx]: Yes, the signup state has one special property: it is transferred from the logged out session into the logged in session. All other state slices are reset. I'm not sure if that property is used anywhere at this moment, but will certainly be useful again soon if it isn't.

This login code prevents clearing the logged out session persisted data after successful login. It only clears the data when some user A is already logged in and then user B logs in.

**Comment by jsnajdr** [comment]
> Any idea why this code wasn't removed when the A/B test was?

I don't know, let's see what @andrewserong has to say here 🙂

**Review by andrewserong — APPROVED** [review]
Thanks @jsnajdr for the ping and the clean-up work in this PR! While the original A/B test wrapped up, we weren't certain in the short-term of the results of the test, and which direction we were ultimately going to go, so it was easier to leave some of this code in, just in case we wanted to re-run the test. It's been long enough now that I'm happy for the rest of this to be removed. And if we do end up wanting to run the test again, I can look into another way to go about it instead of inserting sign up code into the Login block.

Nice change making `rebootAfterLogin` an async function instead, too, it feels cleaner!

**Comment by jsnajdr** [comment]
Thanks @sgomes and @andrewserong for reviewing! Now the door is open to land the Login entrypoint PR.

**Pull request merged**